### ISSUE TITLE
Why System.Uri discards RTL/LTR chars?

### ISSUE THREAD

**Issue by FDUdannychen** [dotnet/corefx#27593]
According to [the document](https://docs.microsoft.com/en-us/dotnet/framework/network-programming/international-resource-identifier-support-in-system-uri), System.Uri supports [IRIs](https://tools.ietf.org/html/rfc3987). In my test code, System.Uri discards RTL/LTR(right-to-left/left-to-right) chars.

```csharp
var codePoints = new[] 
{
    0xE0000, 0xE0FFF, 0xE000, 0xF8FF,                           //some other chars work
    0x200F, 0x200E, 0x202E, 0x202D, 0x202C, 0x202B, 0x202A      //RTL LTR chars discarded
};

foreach (var cp in codePoints)
{
    var uri = new Uri($"http://localhost/{char.ConvertFromUtf32(cp)}");
    if (uri.AbsolutePath.Equals("/"))
    {
        Console.WriteLine($"0x{cp.ToString("X4")} is discarded");
    }
}
```
Am I missing anything?

**Comment by danmosemsft** [comment]
@rmkerr

**Comment by davidsh** [comment]
This could be fixed now as a result of https://github.com/dotnet/corefx/pull/26022

**Comment by rmkerr** [comment]
Yep! As David mentioned this is fixed in the latest 2.1 preview release. There's a surprising amount of ambiguity on the treatment of bidirectional control characters in the IRI spec, but we finally took a look at it and settled on the behavior of leaving them in.

Thanks for the report!

**Comment by FDUdannychen** [comment]
@danmosemsft @davidsh @rmkerr Thank you guys for your effort. Can you share when will 2.1 be released?
